src/sentry/pre_process_forwarder.py

```diff
@@ -0,0 +1,134 @@
+        "default.topic.config": {
+            "auto.offset.reset": "earliest"  # TODO RaduW check what we want to do earliest/latest/error
+        },
+    }
+
+    return kafka.Consumer(consumer_configuration)
+
+
+def run_pre_process_forwarder(
+    commit_batch_size, consumer_group, consumer_type, max_batch_time_seconds
+):
+    """
+    Entry point for pre_process_forwarder, injects dependencies in the internal runner (so that the
+    internal runner can be easily tested)
+    """
+    return _run_pre_process_forwarder_internal(
+        commit_batch_size,
+        consumer_group,
+        consumer_type,
+        max_batch_time_seconds=max_batch_time_seconds,
+        settings=settings,  # by default use the project settings
+        is_shutdown_requested=lambda: False,  # by default run forever (or until keyboard interrupt)
+    )
+
+
+def _run_pre_process_forwarder_internal(
+    commit_batch_size,
+    consumer_group,
+    consumer_type,
+    max_batch_time_seconds,
+    settings,
+    is_shutdown_requested,
+):
+    """
+    Does the pre process forwarder job
+    """
+    logger.debug("Starting pre-process-forwarder...")
+    consumer = _create_consumer(consumer_group, consumer_type, settings)
+
+    consumer.subscribe([ConsumerType.get_topic_name(consumer_type, settings)])
+
+    try:
+        while not (is_shutdown_requested()):
+            # get up to commit_batch_size messages
+            messages = consumer.consume(
+                num_messages=commit_batch_size, timeout=max_batch_time_seconds
+            )
+
+            for message in messages:
+                message_error = message.error()
+                if message_error is not None:
+                    logger.error(
+                        "Received message with error on %s, error:'%s'",
+                        consumer_type,
+                        message_error,
+                    )
+                    raise ValueError(
+                        "Bad message received from consumer", consumer_type, message_error
+                    )
+
+                message = msgpack.unpackb(message.value(), use_list=False)
+                body = message["payload"]
+                start_time = float(message["start_time"])
+                event_id = message["event_id"]
+                project_id = message["project_id"]
+
+                cache_key = cache_key_from_project_id_and_event_id(
+                    project_id=project_id, event_id=event_id
+                )
+                cache_timeout = 3600
+                default_cache.set(cache_key, body, cache_timeout, raw=True)
+                preprocess_event.delay(
+                    cache_key=cache_key, start_time=start_time, event_id=event_id
+                )
+
+            if len(messages) > 0:
+                # we have read some messages in the previous consume, commit the offset
+                consumer.commit(asynchronous=False)
```

Comment:
I think we are ok, if commit fails we will try to commit at next batch (but probably we would fail when we try to read the next message). At worst we would re read the events but in this case the deduplication protection should stop double processing.